python/feathub/feature_views/sliding_feature_view.py

```diff
@@ -270,46 +277,88 @@ def _validate(features: Sequence[Union[str, Feature]]) -> None:
 
         # Per-row transform features listed before the first SlidingWindowTransform
         # feature.
-        per_row_transform_feature_names: Set[str] = set()
-        sliding_window_transforms_group_by_keys: Set[str] = set()
+        pre_sliding_per_row_feature_names: Set[str] = set()
+
+        # Per-row transform features listed after the first SlidingWindowTransform
+        # feature.
+        post_sliding_per_row_features: Set[Feature] = set()
+
+        sliding_feature_names = set()
+        sliding_features_group_by_key: Set[str] = set()
+        sliding_features_variables: Set[str] = set()
 
         for feature in features:
             if isinstance(feature, str):
-                if len(sliding_window_transforms_group_by_keys) == 0:
-                    per_row_transform_feature_names.add(feature)
+                if len(sliding_features_group_by_key) == 0:
+                    pre_sliding_per_row_feature_names.add(feature)
                 continue
```

Comment:
Should we try to handle string transformations after sliding window transform?

Reply:
String transformations can only refer to a feature from the source table. Thus, there should be no string transformation after the sliding window transform.

Reply:
I agree with it that there should be no string transformation after the sliding window transform. And if such cases appeared, FeatHub should throw exceptions instead of ignoring it.

Reply:
The validation is now done after the string feature is resolved. It should not be a problem now.